vision/standard_object/odlc_text_detection.py

```diff
@@ -0,0 +1,132 @@
+"""
+Method for reading the text on the odlc standard objects.
+"""
+
+from typing import Any, Dict, List
+
+import sys
+import cv2
+import numpy as np
+import numpy.typing as npt
+import pytesseract
+
+from vision.common.bounding_box import ObjectType, BoundingBox
+
+# from bounding_box import ObjectType, BoundingBox  # A local file is required when testing
+
+
+def text_detection_pre_processing(unprocessed_img: npt.NDArray[np.uint8]) -> npt.NDArray[np.uint8]:
+    """
+    This does the preprocessing effects on the passed image needed to make text detection easier.
+
+    Parameters
+    ----------
+    img: npt.NDArray[np.uint8]
+        The image that you want to do the preprocessing on.
+
+    Returns
+    ----------
+    final_img: pt.NDArray[np.uint8]
+        The image after all of the preprocessing effects have been run.
+    """
+
+    # Contrast and brightness
+    contrast_img: npt.NDArray[np.uint8] = cv2.convertScaleAbs(unprocessed_img, alpha=1.25, beta=-30)
+
+    # Grayscale
+    grayscale_img: npt.NDArray[np.uint8] = cv2.cvtColor(contrast_img, cv2.COLOR_RGB2GRAY)
+
+    # Blur
+    blured_img: npt.NDArray[np.uint8] = cv2.medianBlur(grayscale_img, ksize=3)
+
+    # Erode and dilate
+    kernel: npt.NDArray[np.uint8] = np.ones((5, 5), np.uint8)
+    eroded_img: npt.NDArray[np.uint8] = cv2.erode(blured_img, kernel=kernel, iterations=1)
+    final_img: npt.NDArray[np.uint8] = cv2.dilate(eroded_img, kernel=kernel, iterations=1)
+
+    return final_img
+
+
+def text_detection(base_img: npt.NDArray[np.uint8]) -> List[str]:
+    """
+    Detects text on the passed image.
+
+    Parameters
+    ----------
+    img: npt.NDArray[np.uint8]
+        The image that you want to detect the text on.
+
+    Returns
+    ----------
+    detected_text: List[str]
+        A list containing each of the individual detected characters in the passed image as strings.
+
+    """
```

Comment:
Number of dashes and remove extra newline before docstring close
```suggestion
    Returns
    -------
    detected_text: List[str]
        A list containing each of the individual detected characters in the passed image as strings.
    """
```